### PR TITLE
Improve existing Cisco C1000 switches and add missing ones

### PR DESCRIPTION
**Pull request by jnovak-netsystemcz**
Added all missing Cisco C1000 and C1000FE switches. Synchronized definition of Cisco C1000 switches.

### CODE REVIEW

**Review by DanSheps — APPROVED** [review]

**Comment by jnovak-netsystemcz** [comment]
@DanSheps Hi. I fixed the issue. Please, reapprove the change.

**Comment by jnovak-netsystemcz** [comment]
@DanSheps I'm sorry, but I don't understand why it is not merged. It is approved and it is not draft. Can I improve the PR somehow?

**Comment by danner26** [comment]
@jnovak-netsystemcz please do me a favor and first sync your forks master branch with the upstream, then merge the current master into this branch. 
You will see merge conflicts. Please fix any issues, run `pre-commit run` on the changes, and then push them for review. Thanks!

File: device-types/Cisco/C1000-48T-4G-L.yaml [hunk 1]
```diff
@@ -7,169 +7,118 @@ comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.co
...
-  - name: Serial Console
+  - name: con0
     type: rj-45
   - name: USB con0
-    type: usb-mini-b
-  - name: USB con1
     type: usb-a
 power-ports:
-  - name: PS-1
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000-48T-4G-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: device-types/Cisco/C1000-8T-2G-L.yaml [hunk 2]
```diff
@@ -7,43 +7,36 @@ comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.co
...
-  - name: Serial Console
+  - name: con0
     type: rj-45
   - name: USB con0
-    type: usb-mini-b
-  - name: USB con1
     type: usb-a
 power-ports:
-  - name: PS-1
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000-8T-2G-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: schema/components.json [hunk 3]
```diff
@@ -171,7 +171,8 @@
                         "usb-3-micro-b",
                         "dc-terminal",
                         "saf-d-grid",
-                        "hardwired"
+                        "hardwired",
```
Comment (danner26) [inline schema/components.json:174]: This file cannot be modified. These are what netbox has upstream by default. **Please remove your changes here and any use of these proposed new ones need to be changed**. PSU types should either reflect one of the already pre-defined names, or `other` if there is not a current type available

If you want a new type added, you will need to be propose a change to the upstream netbox repo. Once added in that repo and pushed to master, we can add them here as well

File: device-types/Cisco/C1000-16P-E-2G-L.yaml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-mini-b
+power-ports:
+  - name: External PSU
+    type: dc-molex-4plug
```
Comment (danner26) [inline device-types/Cisco/C1000-16P-E-2G-L.yaml:16]: This is not a supported type at the moment. You can use `dc-terminal` or `other`, or any other that is correct and supported.

File: device-types/Cisco/C1000-16T-E-2G-L.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-mini-b
+power-ports:
+  - name: External PSU
+    type: dc-molex-4plug
```
Comment (danner26) [inline device-types/Cisco/C1000-16T-E-2G-L.yaml:16]: This is not a supported type at the moment. You can use `dc-terminal` or `other`, or any other that is correct and supported.

File: device-types/Cisco/C1000-8FP-E-2G-L.yaml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-a
+power-ports:
+  - name: External PSU
+    type: dc-molex-4plug
```
Comment (danner26) [inline device-types/Cisco/C1000-8FP-E-2G-L.yaml:16]: This is not a supported type at the moment. You can use `dc-terminal` or `other`, or any other that is correct and supported.

File: device-types/Cisco/C1000-8P-E-2G-L.yaml [hunk 7]
```diff
@@ -3,37 +3,56 @@ manufacturer: Cisco
...
-  - name: USB Console
-    type: usb-mini-b
+  - name: USB con0
+    type: usb-a
 power-ports:
-  - name: ExternalPS
-    type: iec-60320-c14
-    maximum_draw: 85
+  - name: External PSU
+    type: dc-molex-4plug
```
Comment (danner26) [inline device-types/Cisco/C1000-8P-E-2G-L.yaml:16]: This is not a supported type at the moment. You can use `dc-terminal` or `other`, or any other that is correct and supported.

File: device-types/Cisco/C1000-8T-E-2G-L.yaml [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-a
+power-ports:
+  - name: External PSU
+    type: dc-molex-4plug
```
Comment (danner26) [inline device-types/Cisco/C1000-8T-E-2G-L.yaml:16]: This is not a supported type at the moment. You can use `dc-terminal` or `other`, or any other that is correct and supported.

File: device-types/Cisco/C1000-16FP-2G-L.yaml [hunk 9]
```diff
@@ -3,52 +3,86 @@ manufacturer: Cisco
...
 comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.com/c/en/us/products/collateral/switches/catalyst-1000-series-switches/nb-06-cat1k-ser-switch-ds-cte-en.html)'
+u_height: 1
 is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-mini-b
+power-ports:
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000-16FP-2G-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: device-types/Cisco/C1000-16P-2G-L.yaml [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.com/c/en/us/products/collateral/switches/catalyst-1000-series-switches/nb-06-cat1k-ser-switch-ds-cte-en.html)'
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-mini-b
+power-ports:
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000-16P-2G-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: device-types/Cisco/C1000-16T-2G-L.yaml [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.com/c/en/us/products/collateral/switches/catalyst-1000-series-switches/nb-06-cat1k-ser-switch-ds-cte-en.html)'
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-mini-b
+power-ports:
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000-16T-2G-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: device-types/Cisco/C1000-24FP-4G-L.yaml [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.com/c/en/us/products/collateral/switches/catalyst-1000-series-switches/nb-06-cat1k-ser-switch-ds-cte-en.html)'
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-a
+power-ports:
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000-24FP-4G-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: device-types/Cisco/C1000-24FP-4X-L.yaml [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.com/c/en/us/products/collateral/switches/catalyst-1000-series-switches/nb-06-cat1k-ser-switch-ds-cte-en.html)'
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-a
+power-ports:
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000-24FP-4X-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: device-types/Cisco/C1000-24P-4G-L.yaml [hunk 14]
```diff
@@ -3,66 +3,115 @@ manufacturer: Cisco
...
 console-ports:
-  - name: Serial Console
+  - name: con0
     type: rj-45
-  - name: USB Console
-    type: usb-mini-b
+  - name: USB con0
+    type: usb-a
 power-ports:
   - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000-24P-4G-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: device-types/Cisco/C1000-24P-4X-L.yaml [hunk 15]
```diff
@@ -3,55 +3,111 @@ manufacturer: Cisco
...
+comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.com/c/en/us/products/collateral/switches/catalyst-1000-series-switches/nb-06-cat1k-ser-switch-ds-cte-en.html)'
 u_height: 1
 is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-a
+power-ports:
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000-24P-4X-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: device-types/Cisco/C1000-24T-4G-L.yaml [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.com/c/en/us/products/collateral/switches/catalyst-1000-series-switches/nb-06-cat1k-ser-switch-ds-cte-en.html)'
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-a
+power-ports:
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000-24T-4G-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: device-types/Cisco/C1000-24T-4X-L.yaml [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.com/c/en/us/products/collateral/switches/catalyst-1000-series-switches/nb-06-cat1k-ser-switch-ds-cte-en.html)'
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-a
+power-ports:
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000-24T-4X-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: device-types/Cisco/C1000-48FP-4G-L.yaml [hunk 18]
```diff
@@ -0,0 +1,220 @@
...
+comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.com/c/en/us/products/collateral/switches/catalyst-1000-series-switches/nb-06-cat1k-ser-switch-ds-cte-en.html)'
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-a
+power-ports:
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000-48FP-4G-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: device-types/Cisco/C1000-48FP-4X-L.yaml [hunk 19]
```diff
@@ -0,0 +1,220 @@
...
+comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.com/c/en/us/products/collateral/switches/catalyst-1000-series-switches/nb-06-cat1k-ser-switch-ds-cte-en.html)'
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-a
+power-ports:
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000-48FP-4X-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: device-types/Cisco/C1000-48P-4G-L.yaml [hunk 20]
```diff
@@ -0,0 +1,220 @@
...
+comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.com/c/en/us/products/collateral/switches/catalyst-1000-series-switches/nb-06-cat1k-ser-switch-ds-cte-en.html)'
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-a
+power-ports:
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000-48P-4G-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: device-types/Cisco/C1000-48P-4X-L.yaml [hunk 21]
```diff
@@ -0,0 +1,220 @@
...
+comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.com/c/en/us/products/collateral/switches/catalyst-1000-series-switches/nb-06-cat1k-ser-switch-ds-cte-en.html)'
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-a
+power-ports:
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000-48P-4X-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: device-types/Cisco/C1000-48T-4X-L.yaml [hunk 22]
```diff
@@ -0,0 +1,124 @@
...
+comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.com/c/en/us/products/collateral/switches/catalyst-1000-series-switches/nb-06-cat1k-ser-switch-ds-cte-en.html)'
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-a
+power-ports:
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000-48T-4X-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: device-types/Cisco/C1000-8FP-2G-L.yaml [hunk 23]
```diff
@@ -0,0 +1,58 @@
...
+comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.com/c/en/us/products/collateral/switches/catalyst-1000-series-switches/nb-06-cat1k-ser-switch-ds-cte-en.html)'
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-a
+power-ports:
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000-8FP-2G-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: device-types/Cisco/C1000-8P-2G-L.yaml [hunk 24]
```diff
@@ -0,0 +1,58 @@
...
+comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.com/c/en/us/products/collateral/switches/catalyst-1000-series-switches/nb-06-cat1k-ser-switch-ds-cte-en.html)'
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-a
+power-ports:
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000-8P-2G-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: device-types/Cisco/C1000FE-24P-4G-L.yaml [hunk 25]
```diff
@@ -0,0 +1,126 @@
...
+comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.com/c/en/us/products/collateral/switches/catalyst-1000-series-switches/nb-06-cat1k-ser-switch-ds-cte-en.html)'
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-a
+power-ports:
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000FE-24P-4G-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: device-types/Cisco/C1000FE-24T-4G-L.yaml [hunk 26]
```diff
@@ -0,0 +1,78 @@
...
+comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.com/c/en/us/products/collateral/switches/catalyst-1000-series-switches/nb-06-cat1k-ser-switch-ds-cte-en.html)'
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-a
+power-ports:
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000FE-24T-4G-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: device-types/Cisco/C1000FE-48P-4G-L.yaml [hunk 27]
```diff
@@ -0,0 +1,222 @@
...
+comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.com/c/en/us/products/collateral/switches/catalyst-1000-series-switches/nb-06-cat1k-ser-switch-ds-cte-en.html)'
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-a
+power-ports:
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000FE-48P-4G-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

File: device-types/Cisco/C1000FE-48T-4G-L.yaml [hunk 28]
```diff
@@ -0,0 +1,126 @@
...
+comments: '[Cisco Catalyst 1000 Series Switches Data Sheet](https://www.cisco.com/c/en/us/products/collateral/switches/catalyst-1000-series-switches/nb-06-cat1k-ser-switch-ds-cte-en.html)'
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: USB con0
+    type: usb-a
+power-ports:
+  - name: Built-in
```
Comment (danner26) [inline device-types/Cisco/C1000FE-48T-4G-L.yaml:15]: Current documentation addresses the PSUs as PSU1, PSU2, etc. Please revert this to PS-1 or PSU1 (my preference is PSU1), or provide documentation where it is called Build-in. Thanks!

**Review by danner26 — CHANGES_REQUESTED** [review]
Please review the requested changes and see my following comment for more information

**Comment by danner26** [comment]
@jnovak-netsystemcz 

First, we cannot modify the components.json schema. That is a reflection of what NetBox will accept by default. Please remove your changes to that file

Why did you change the power port names to Built-in? I havent worked with 1000 series Cisco devices, so I might just be off base. The cisco docs I am familiar with call them PSU1, PSU2, etc. If there is documentation or writing on the unit that says Built-in rather than PS-1 or PSU1 **please provide that**. Otherwise please revert those changes to PS-1 or PSU1 (my preference is PSU1)

File: device-types/Cisco/C1000-48T-4G-L.yaml (hunk 1 above)
Comment (jnovak-netsystemcz) [reply device-types/Cisco/C1000-48T-4G-L.yaml:15]: I'm sorry for this change. We are used to name PSU/PSU1/... power supplies which are field replaceable and we are numbering them when there is possible to insert more than one unit.
When we don't want to use Build-in (some of units use external ones), I propose to name it just PSU. Is it OK.

File: schema/components.json (hunk 3 above)
Comment (jnovak-netsystemcz) [reply schema/components.json:174]: I was not aware the file can't be changed - I'm new to netbox...

In this case dc-terminal is fine for me.

Comment (danner26) [reply schema/components.json:174]: Perfect, go ahead and revert the changes here, then update your files accordingly. Thank you!

**Comment by danner26** [comment]
@jnovak-netsystemcz Can you please change your power supply definitions from `PSU` to `PSU1`?

**Comment by jnovak-netsystemcz** [comment]
As I wrote, there is no other PSU than the one so PSU<number> makes no sense, but OK. Just a second...

**Comment by danner26** [comment]
Great, I will go through and check everything
My idea behind that is in an effort to standardize the data, all PSU's should have a number suffix even if they only support 1

**Comment by jnovak-netsystemcz** [comment]
I see. Thank you.

**Review by danner26 — APPROVED** [review]

**Pull request merged**